daemon/cmd/daemon.go

```diff
@@ -401,10 +401,13 @@ func NewDaemon(ctx context.Context, cancel context.CancelFunc, epMgr *endpointma
 	// detection, might disable BPF NodePort and friends. But this is fine, as
 	// the feature does not influence the decision which BPF maps should be
 	// created.
-	isKubeProxyReplacementStrict, err := initKubeProxyReplacementOptions()
-	if err != nil {
-		log.WithError(err).Error("unable to initialize Kube proxy replacement options")
-		return nil, nil, fmt.Errorf("unable to initialize Kube proxy replacement options: %w", err)
+	var isKubeProxyReplacementStrict bool
+	if !option.Config.DryMode {
+		isKubeProxyReplacementStrict, err = initKubeProxyReplacementOptions()
```

Comment:
We needn't skip the initialization of kpr options altogether - just the probing part.

Reply:
Yep I should've been more careful and not leave the result of exploration in. I'll remove this and make sure initKubeProxyReplacement() works in DryMode (e.g. no probing).